vscode/vscode/vscode.ml

```diff
@@ -2185,16 +2370,16 @@ module Window = struct
     -> Disposable.t
     [@@js.global "vscode.window.setStatusBarMessage"]
 
-  val withProgress :
-       options:ProgressOptions.t
-    -> task:
-         (progress:Progress.t -> token:CancellationToken.t -> Ojs.t Promise.t)
-    -> Ojs.t Promise.t
+  val withProgress : options:ProgressOptions.t -> task:Ojs.t -> Ojs.t
     [@@js.global "vscode.window.withProgress"]
 
-  let withProgress ~options ~task =
-    let task ~progress ~token = Obj.magic (task ~progress ~token) in
-    Obj.magic (withProgress ~options ~task)
+  let withProgress (type a) (module T : Js.T with type t = a) ~options ~task :
```

Comment:
I might be not understanding the point of this, but why do we need to pass this `T` module`? we aren't doing anything with its fields from what I can see.

Reply:
It's used in a gen_js_api extension for converting `T.t Promise.t` types in the function body.